server/src/main/java/org/opensearch/cluster/metadata/MetadataUpdateSettingsService.java

```diff
@@ -111,6 +112,11 @@ public MetadataUpdateSettingsService(
         this.indicesService = indicesService;
         this.shardLimitValidator = shardLimitValidator;
         this.awarenessReplicaBalance = awarenessReplicaBalance;
+
+        /*
+         * Task will get retried from { @link TransportClusterManagerNodeAction}
+         */
+        ClusterManagerTaskThrottler.registerThrottlingKey(UPDATE_SETTING_CLUSTER_MANAGER_TASK_KEY, true);
```

Comment:
do we need the value bool, when should it be false?

Reply:
If task is not extended by `TransportClusterManagerNodeAction` and don't have explicit retries on throttling, then we can register that task key with false value. One example could be `ShardStateAction`. 
Idea is throttler will not allow to set the throttling threshold for such tasks, else data node can fail with throttling exception. 

With this we can get aggregated count of those task from new map which we are maintaining in master side.